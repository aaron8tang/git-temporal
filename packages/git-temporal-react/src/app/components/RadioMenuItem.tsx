import React from 'react';
import { style } from 'app/styles';
import { TestProps } from 'app/interfaces';
import { Selectable } from 'app/components/Selectable';
import { ToggleButton } from 'app/components/ToggleButton';

export interface RadioMenuItemProps {
  // The children are the menu content
  children: string | JSX.Element | JSX.Element[];
  isSelected: boolean;
  onClick: (evt) => void;
  style?: string | object;
  testId?: string;
  disabled?: boolean;
}

const toggleButtonStyle = {
  _extends: 'selectable',
  display: 'inline-block',
  width: 15,
  height: 15,
  borderRadius: 7,
  marginRight: 10,
};

export class RadioMenuItem extends React.Component<
  RadioMenuItemProps & TestProps
> {
  render() {
    return (
      <Selectable
        style={style(this.props.style)}
        onClick={this.props.onClick}
        disabled={this.props.disabled}
        testId={this.props.testId}
      >
        <div>
          <ToggleButton
            style={toggleButtonStyle}
            isSelected={this.props.isSelected}
            onClick={this.props.onClick}
          />
          {this.props.children}
        </div>
      </Selectable>
    );
  }
}
