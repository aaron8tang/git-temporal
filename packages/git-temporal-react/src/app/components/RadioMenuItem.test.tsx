import React from 'react';
import { shallow } from 'enzyme';

import { RadioMenuItem } from './RadioMenuItem';

describe('components/RadioMenuItem', () => {
  describe('when rendered with isSelected=true', () => {
    const clickFn = jest.fn();
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <RadioMenuItem isSelected={true} onClick={clickFn}>
          This is a radio menu item
        </RadioMenuItem>
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('when rendered with isSelected=false', () => {
    const clickFn = jest.fn();
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <RadioMenuItem isSelected={false} onClick={clickFn}>
          This is a radio menu item
        </RadioMenuItem>
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
