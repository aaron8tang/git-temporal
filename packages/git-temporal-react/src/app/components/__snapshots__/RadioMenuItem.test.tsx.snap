// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RadioMenuItem when rendered with isSelected=false it should match snapshot 1`] = `
<Selectable
  onClick={[MockFunction]}
  style={Object {}}
>
  <div>
    <ToggleButton
      isSelected={false}
      onClick={[MockFunction]}
      style={
        Object {
          "borderRadius": 7,
          "display": "inline-block",
          "height": 15,
          "marginRight": 10,
          "width": 15,
        }
      }
    />
    This is a radio menu item
  </div>
</Selectable>
`;

exports[`components/RadioMenuItem when rendered with isSelected=true it should match snapshot 1`] = `
<Selectable
  onClick={[MockFunction]}
  style={Object {}}
>
  <div>
    <ToggleButton
      isSelected={true}
      onClick={[MockFunction]}
      style={
        Object {
          "borderRadius": 7,
          "display": "inline-block",
          "height": 15,
          "marginRight": 10,
          "width": 15,
        }
      }
    />
    This is a radio menu item
  </div>
</Selectable>
`;
